txnkv/transaction/pessimistic.go

```diff
@@ -227,6 +227,7 @@ func (action actionPessimisticLock) handleSingleBatch(c *twoPhaseCommitter, bo *
 		// tikv default will wait 3s(also the maximum wait value) when lock error occurs
 		startTime = time.Now()
 		msBeforeTxnExpired, err := c.store.GetLockResolver().ResolveLocks(bo, 0, locks)
+		defer c.store.GetLockResolver().ResolveLocksDone(c.startTS)
```

Comment:
It's in a retry loop here. It seems we'd better protect it from being deferred more than once or do the call before entering the next loop? The same question to the prewrite part.